micro-deps/micro-deps-spring-config/src/main/java/com/ofg/infrastructure/discovery/AsyncRegistrationServiceDiscoveryFactoryBean.java

```diff
@@ -0,0 +1,43 @@
+package com.ofg.infrastructure.discovery;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.curator.x.discovery.ServiceDiscovery;
+import org.springframework.beans.factory.FactoryBean;
+
+public class AsyncRegistrationServiceDiscoveryFactoryBean implements FactoryBean<ServiceDiscovery>{
+
+    private final ServiceDiscovery serviceDiscovery;
+
+    private ExecutorService executorService = Executors.newFixedThreadPool(1);
```

Comment:
Just start a `Thread`, you don't need a pool for just one task. Which you, BTW, don't shut down.